plugins/woocommerce/client/blocks/assets/js/blocks/payment-method-icons/edit.tsx

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useBlockProps, InspectorControls } from '@wordpress/block-editor';
+import { PanelBody, RangeControl } from '@wordpress/components';
+import { getPaymentMethods } from '@woocommerce/blocks-registry';
+
+const CardPreview = ( {
+	type,
+	icon,
+}: {
+	type: string | undefined;
+	icon: string | undefined;
+} ) => {
+	let CardIcon = null;
+
+	if ( type && icon ) {
+		CardIcon = (
+			<div className="wp-block-woocommerce-payment-method-icons__item">
+				<span
+					className="wp-block-woocommerce-payment-method-icons__icon"
+					style={ {
+						backgroundImage: `url(${ icon })`,
+					} }
+					role="img"
+					aria-label={ type }
+				/>
+			</div>
+		);
+	}
+
+	return CardIcon;
+};
+
+const Edit = ( {
+	attributes,
+	setAttributes,
+}: {
+	attributes: {
+		numberOfIcons: number;
+	};
+	setAttributes: ( attributes: Record< string, unknown > ) => void;
+} ) => {
+	const blockProps = useBlockProps();
+	const paymentMethodData = getPaymentMethods();
+	const wooPaymentMethods =
+		paymentMethodData?.woocommerce_payments?.content?.props?.upeMethods;
+	const { numberOfIcons } = attributes;
+
+	if ( wooPaymentMethods ) {
+		const wcSettings = window.wcSettings as {
+			cardIcons?: Record< string, { icon: string } >;
+		};
+		const cardIcons = wcSettings?.cardIcons || {};
+		const availableCardIcons = Object.keys( cardIcons )
+			.map( ( type ) => {
+				if ( ! cardIcons[ type ] || ! cardIcons[ type ].icon ) {
+					return null;
+				}
+				return {
+					type,
+					icon: cardIcons[ type ].icon,
+				};
+			} )
+			.filter( Boolean );
```

Comment:
What about converting `map + filter` (two loops) into `reduce` (single loop)?

```suggestion
const availableCardIcons = Object.keys( cardIcons ).reduce( ( acc, type ) => {
    if ( ! cardIcons[ type ] || !cardIcons[ type ].icon) {
        return acc;
    }
    
    return [ ...acc, {
        type,
        icon: cardIcons[ type ].icon,
    } ];
}, [] );
```

Reply:
Looks good! Done in https://github.com/woocommerce/woocommerce/pull/57181/commits/294348556594e7b579e713201516d2f8c4336aeb.